packages/apps/human-app/frontend/src/shared/contexts/wallet-connect.tsx

```diff
@@ -102,52 +72,60 @@ export function WalletConnectProvider({
     }
   }, [web3ProviderMutation]);
 
+  const openModal = useCallback(async () => {
+    await open();
+  }, [open]);
+
+  const signMessage = useCallback(
+    async (message: string) => {
+      if (web3ProviderMutation.data) {
+        try {
+          const signature =
+            await web3ProviderMutation.data.signer.signMessage(message);
+          return signature;
+        } catch (error) {
+          throw new JsonRpcError(error);
+        }
+      }
+      return Promise.resolve(undefined);
```

Comment:
You don't need this line in `async` function: it resolves with `undefined` if other is not specified